sound/soc/sof/pm.c

```diff
@@ -312,6 +312,13 @@ int snd_sof_prepare(struct device *dev)
 	/* will suspend to S3 by default */
 	sdev->system_suspend_target = SOF_SUSPEND_S3;
 
+	/*
+	 * if the firmware is crashed then we try to aim for S3 to reboot the
+	 * firmware
+	 */
+	if (sdev->fw_state == SOF_FW_CRASHED)
+		return 0;
```

Comment:
why is this important @ujfalusi. prepapre really only prepares for the target state for the system no?

Reply:
@ranj063, probably. We will force `SOF_SUSPEND_S3`, see line 313